beacon-chain/rpc/beacon/attestations.go

```diff
@@ -134,44 +147,46 @@ func (bs *Server) ListIndexedAttestations(
 			NextPageToken:       strconv.Itoa(0),
 		}, nil
 	}
-
-	// We use the retrieved committees for the epoch to convert all attestations
+	// We use the retrieved committees for the block root to convert all attestations
 	// into indexed form effectively.
+	mappedAttestations := mapAttestationToBlockRoot(ctx, attsArray)
 	indexedAtts := make([]*ethpb.IndexedAttestation, numAttestations, numAttestations)
-	for i := 0; i < len(atts); i++ {
-		att := atts[i]
-		epoch := helpers.SlotToEpoch(att.Data.Slot)
-		attState, err := bs.StateGen.StateByRoot(ctx, bytesutil.ToBytes32(att.Data.BeaconBlockRoot))
-		if err != nil {
-			return nil, status.Errorf(
-				codes.Internal,
-				"Could not retrieve state for attestation data block root %v: %v",
-				att.Data.BeaconBlockRoot,
-				err,
-			)
-		}
-		activeIndices, err := helpers.ActiveValidatorIndices(attState, epoch)
-		if err != nil {
-			return nil, status.Errorf(
-				codes.Internal,
-				"Could not retrieve active validator indices for epoch %d: %v",
-				epoch,
-				err,
-			)
+	for br, atts := range mappedAttestations {
+		var attState *stateTrie.BeaconState
+		if !featureconfig.Get().DisableNewStateMgmt {
+			attState, err = bs.StateGen.StateByRoot(ctx, br)
+			if err != nil {
+				return nil, status.Errorf(
+					codes.Internal,
+					"Could not retrieve state for attestation data block root %v: %v",
+					br,
+					err,
+				)
+			}
+		} else {
+			attState, err = bs.BeaconDB.State(ctx, br)
+			if err != nil {
+				return nil, status.Errorf(
+					codes.Internal,
+					"Could not retrieve state for attestation data block root %v: %v",
+					br,
+					err,
+				)
+			}
 		}
-		seed, err := helpers.Seed(attState, epoch, params.BeaconConfig().DomainBeaconAttester)
-		if err != nil {
-			return nil, status.Errorf(
-				codes.Internal,
-				"Could not seed for epoch %d: %v",
-				epoch,
-				err,
-			)
+		for i := 0; i < len(atts); i++ {
+			att := atts[i]
+			committee, err := helpers.BeaconCommitteeFromState(attState, att.Data.Slot, att.Data.CommitteeIndex)
+			if err != nil {
+				return nil, status.Errorf(
+					codes.Internal,
+					"Could not retrieve committees from state %v",
+					err,
+				)
+			}
+			idxAtt := attestationutil.ConvertToIndexed(ctx, att, committee)
+			indexedAtts[i] = idxAtt
```

Comment:
This uses `len(atts)` for `i` but isn't `indexedAtts` `numAttestations` in length?

Reply:
thanks fo catching this